test/strategies/MgvOrder.t.sol

```diff
@@ -272,481 +333,489 @@ contract MangroveOrder_Test is MangroveTest {
 
-  function test_underprovisioned_order_logs_properly() public {
-    IOrderLogic.TakerOrder memory buyOrder = IOrderLogic.TakerOrder({
-      outbound_tkn: base,
-      inbound_tkn: quote,
-      fillOrKill: false,
-      fillWants: true,
-      takerWants: 2 ether,
-      takerGives: 0.26 ether,
-      restingOrder: true,
-      pivotId: 0,
-      expiryDate: 0
-    });
-    deal($(quote), $(this), 0.25 ether);
-    vm.expectRevert("mgvOrder/transferInFail");
-    mgo.take{value: 0.1 ether}(buyOrder);
+  function test_only_offer_owner_can_update_offer() public {
+    vm.expectRevert("AccessControlled/Invalid");
+    vm.prank(freshAddress());
+    mgo.updateOffer(quote, base, 1 ether, 2000 ether, cold_buyResult.offerId, cold_buyResult.offerId);
   }
 
-  function test_caller_unable_to_receive_eth_makes_failing_resting_order_throw() public {
-    TestSender buy_taker = new TestSender();
-    vm.deal($(buy_taker), 1 ether);
+  //////////////////////////////
+  /// Gas requirements tests ///
+  //////////////////////////////
 
-    deal($(quote), $(buy_taker), 10 ether);
-    buy_taker.refuseNative();
+  function test_mockup_routing_gas_cost() public {
+    SimpleRouter router = SimpleRouter(address(mgo.router()));
+    // making quote balance hot to mock taker's transfer
+    quote.transfer($(mgo), 1);
 
-    vm.startPrank($(buy_taker));
-    TransferLib.approveToken(quote, $(mgo.router()), type(uint).max);
-    vm.stopPrank();
+    vm.prank($(mgo));
+    uint g = gasleft();
+    uint pushed = router.push(quote, address(this), 1);
+    uint push_cost = g - gasleft();
+    assertEq(pushed, 1, "Push failed");
 
-    IOrderLogic.TakerOrder memory buyOrder = IOrderLogic.TakerOrder({
-      outbound_tkn: base,
-      inbound_tkn: quote,
-      fillOrKill: false,
-      fillWants: true,
-      takerWants: 1 ether,
-      takerGives: 0.13 ether,
-      restingOrder: true,
-      pivotId: 0,
-      expiryDate: 0
+    vm.prank($(mgo));
+    g = gasleft();
+    uint pulled = router.pull(base, address(this), 1, true);
+    uint pull_cost = g - gasleft();
+    assertEq(pulled, 1, "Pull failed");
+
+    console.log("Gas cost: %d (pull: %d g.u, push: %d g.u)", pull_cost + push_cost, pull_cost, push_cost);
+  }
+
+  function test_mockup_offerLogic_gas_cost() public {
+    (MgvLib.SingleOrder memory sellOrder, MgvLib.OrderResult memory result) = mockSellOrder({
+      takerGives: 0.5 ether,
+      takerWants: 1991 ether / 2,
+      partialFill: 2,
+      base_: base,
+      quote_: quote,
+      makerData: ""
     });
-    /// since `buy_taker` throws on `receive()`, this should fail.
-    vm.expectRevert("mgvOrder/refundFail");
-    vm.prank($(buy_taker));
-    // complete fill will not lead to a resting order
-    mgo.take{value: 0.1 ether}(buyOrder);
+    // mock up mgv taker transfer to maker contract
+    base.transfer($(mgo), 0.5 ether);
+    sellOrder.offerId = cold_buyResult.offerId;
+    vm.prank($(mgv));
+    _gas();
+    mgo.makerExecute(sellOrder);
```

Comment:
I read your tests correctly it seems we are including the cost of reading offer information in the estimate - but I would expect offer information to already have been read by the core protocol when it invokes makerExecute?

Reply:
Right I updated this